### ISSUE TITLE
Excluding tests from build is bad

### ISSUE THREAD

**Issue by claudiu-cristea** [drush-ops/drush#4104]
**Describe the bug**

In https://github.com/drush-ops/drush/commit/046e5c1cf8c520b988e73e5ccb82cf6eebac2600 some directories were removed from being packaged in the dist. This is bad because if a PR contains tests (and it should in most of the cases), composer fails to apply the PR's patch in a project where Drush is installed as dist.

**To Reproduce**

`composer.json`:

```json
    ...
    "require": {
        ...
        "drush/drush": "~9.7",
        ...
    },
    ...
    "extra": {
        ...
        "patches": {
            ...
            "drush/drush": {
                "PR 4103": "https://patch-diff.githubusercontent.com/raw/drush-ops/drush/pull/4103.diff"
            },
            ...
        },
    },
    ...
```

**Expected behavior**

The patch applies cleanly if there are no organic conflicts.

**Actual behavior**

The patch doesn't apply.

**Workaround**

Require `drush/drush:dev-master`.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 9.x
| Drupal version? | 8.x
| PHP version     | 5.6/7.1
| OS?             | Mac/Linux/Windows

**Additional information**

Please revert this commit. The problems trying to solve are minor but without we a have better DX.

**Comment by pfrenssen** [comment]
I agree this can be annoying but the test files do not really belong in the `dist` build. I wonder if there a way that we can specify to download the `source` build for a single package.

**Comment by greg-1-anderson** [comment]
Yes, I think that you should install Drush via `source` rather than `dist` if you want to patch.

If you only have one patch to apply, then you can do this:

```
"drush:drush": "dev-install-modules-via-update",
```
This works directly if the branch you want to install from is in `drush-ops/drush`. If it isn't, you may add an entry to `repositories` to install Drush from your fork. If you install Drush from any `dev` branch, then you will get a `source` install rather than a `dist` install, so you could add more patches if you wanted (assuming that none of the patches you want conflict with each other).

I personally do not feel inclined to re-add the tests directory to the dist builds. It probably wouldn't hurt too much to have them there, but they do not really belong there, and patching is not a `dist` use-case.
